src/ray/core_worker/core_worker.cc

```diff
@@ -1375,15 +1375,14 @@ Status CoreWorker::GetNamedActorHandle(const std::string &name,
   RAY_CHECK_OK(gcs_client_->Actors().AsyncGetByName(
       name, [this, &actor_id, name, ready, m, cv](
                 Status status, const boost::optional<gcs::ActorTableData> &result) {
-        if (!status.ok()) {
-          RAY_LOG(INFO) << "Failed to look up actor with name: " << name;
-          // Use a NIL actor ID to signal that the actor wasn't found.
-          actor_id = ActorID::Nil();
-        } else {
-          RAY_CHECK(result);
+        if (status.ok() && result) {
```

Comment:
This is modified because this will cause check fails
```
DetachedActor.options(name="blah").remote()
a = ray.get_actor(name="blah")
```
It is because when get_actor is called, the detached actor hasn't been created yet as it is asynchronous. So we should loosen this check.